src/shared/ip-protocol-list.c

```diff
@@ -0,0 +1,40 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <netinet/in.h>
+#include <string.h>
+
+#include "ip-protocol-list.h"
+#include "macro.h"
+
+static const struct ip_protocol_name* lookup_ip_protocol(register const char *str, register GPERF_LEN_TYPE len);
+
+#include "ip-protocol-from-name.h"
+#include "ip-protocol-to-name.h"
+
+const char *ip_protocol_to_name(int id) {
+
+        if (id < 0)
+                return NULL;
+
+        if (id >= (int) ELEMENTSOF(ip_protocol_names))
```

Comment:
not that it matters, but I'd always cast the smaller type to the large type rather than the other way round here. i.e. since `ELEMENTSOF()` returns `size_t` and that is usually larger than `int` (at least on 64bit archs it is), hence cast `id` to `size_t` here (which is safe, after all we checked further up that it is not negative)